src/fp.rs

```diff
@@ -279,17 +279,17 @@ fn modp(x: u128, p: u128) -> u128 {
     z.wrapping_add(m & p)
 }
 
-pub(crate) const FP32: FieldParameters = FieldParameters {
+pub(crate) const FP32: crate::fp64::FieldParameters64 = crate::fp64::FieldParameters64 {
     p: 4293918721, // 32-bit prime
     mu: 17302828673139736575,
-    r2: 1676699750,
-    g: 1074114499,
```

Comment:
Changing the generator breaks interop, which is why I think a test is failing. Why change this? Does this benefit performance?

Reply:
no, it shouldn't change. I going to revert it to the original value.

Reply:
I think this should change, but to a different value. This is because `g` needs to be the Montgomery representation of the generator, and changing big R changes the mapping from residue representation to Montgomery representation. I added the following to `documentation/field_parameters.sage`, to recalculate these parameters with big R equal to 2^64.

```sage
    Field(
        "Field32, u64",
        2 ^ 20 * 4095 + 1,
        3925978153,
        2 ^ 64,
        2 ^ 64,
    ),
```

I get the following output.

```
Field32, u64
p: 4293918721
mu: 17302828673139736575
r2: 2564090464
g: 45203785
num_roots: 20
bit_mask: 4294967295
roots:
[266338049,
 4027580672,
 1578227120,
 3779916562,
 273724820,
 3424176308,
 1517016107,
 1358722360,
 1304575291,
 4258821146,
 4139968002,
 1029526548,
 717934416,
 1564021295,
 3160561122,
 1320076770,
 3893444443,
 1375455849,
 849387224,
 3198638617,
 45203785]
```

Reply:
Correct, it has to change due to Montgomery representation, but keeping the same integer value. Making changes now.